source/module_hamilt_general/module_xc/NCLibxc/NCLibxc.cpp

```diff
@@ -0,0 +1,302 @@
+        postlibxc_lda(xc_id, rho_up, rho_down, e, v1, v2, f1, f2, f3);
+
+        std::vector<double> Eeff(num_points, 0.0);
+        std::vector<Matrix2x2> Veff(num_points, {{{0.0, 0.0}, {0.0, 0.0}}});
+
+        for (size_t i = 0; i < num_points; ++i)
+        {
+            Eeff[i] = e[i] + 0.5 * (m_omega[i] / n[i]) * (v1[i] - v2[i]);
+
+            Matrix2x2 pauli_matrix = construct_pauli_matrix(x, y, z);
+            Matrix2x2 term1 = scalar_multiply(((v1[i] - v2[i]) + 0.25 * m_omega[i] * (f1[i] + f3[i] - 2 * f2[i])), pauli_matrix);
+            Matrix2x2 term2 = scalar_multiply((0.5 * (v1[i] + v2[i]) + 0.25 * m_omega[i] * (f1[i] - f3[i])), identity_matrix());
+            Veff[i] = add(term1, term2);
+
+            E[i] += Eeff[i]*w;
+            V[i] = add(V[i], scalar_multiply(w, Veff[i]));
+        }
+    }
+
+    return {E, V};
+}
+
+std::pair<std::vector<double>, std::vector<Matrix2x2>> NCLibxc::lda_lc(int xc_id, const std::vector<double>& n, 
+                                                              const std::vector<double>& mx, const std::vector<double>& my, const std::vector<double>& mz)
+{
+    size_t num_points = n.size();
+    std::vector<double> E(num_points, 0.0);
+    std::vector<Matrix2x2> V(num_points, {{{0.0, 0.0}, {0.0, 0.0}}});
+    std::vector<double> m_mod(num_points, 0.0);
+
+    // compute |m|
+    for (size_t i = 0; i < num_points; ++i)
+    {
+        m_mod[i] = std::sqrt(mx[i] * mx[i] + my[i] * my[i] + mz[i] * mz[i]);
+    }
+
+    std::vector<double> rho_up(num_points, 0.0);
+    std::vector<double> rho_down(num_points, 0.0);
+
+    for (size_t i = 0; i < num_points; ++i)
+    {
+        rho_up[i] = (n[i] + m_mod[i]) / 2.0;
+        rho_down[i] = (n[i] - m_mod[i]) / 2.0;
+    }
+
+    std::vector<double> e(num_points, 0.0), v1(num_points, 0.0), v2(num_points, 0.0), f1(num_points, 0.0), f2(num_points, 0.0), f3(num_points, 0.0);
+    postlibxc_lda(xc_id, rho_up, rho_down, e, v1, v2, f1, f2, f3);
+
+    for (size_t i = 0; i < num_points; ++i)
+    {
+        E[i] = e[i];
+
+        double x = mx[i] / m_mod[i];
+        double y = my[i] / m_mod[i];
+        double z = mz[i] / m_mod[i];
+
+        Matrix2x2 pauli_matrix = construct_pauli_matrix(x, y, z);
+        Matrix2x2 term1 = scalar_multiply(0.5 * (v1[i] - v2[i]), pauli_matrix);
+        Matrix2x2 term2 = scalar_multiply(0.5 * (v1[i] + v2[i]), identity_matrix());
+        V[i] = add(term1, term2);
+    }
+
+    return {E, V};
+}
+
+///////////////////////////////////////////////////////////////////////////////////
+// print message and citation of the program
+// Function to print message and citation
+void NCLibxc::print_NCLibxc()
+{
+    std::ofstream log_file("NCLibxc.log", std::ios::out | std::ios::app);
+    if (log_file.is_open())
+    {
+        log_file << "You are using the multi-collinear approach implemented by Xiaoyu Zhang. Please cite:\n";

```

Comment:
which paper should be site?

Reply:
This paper hasn't been well prepared. This log is only for tests. We are going to add citation  requests after papers published.